File: Data Structures & Algorithms/duplicate-integer/submission-5.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
    set<int> se;
    for (int &x: nums){
        se.insert(x);
    }
    if (se.size() == nums.size()){
        return false;
    }else{
        return true;
    }
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-5.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        map = {}
        for i in range(len(nums)):
            if map.get(nums[i], -1) != -1 and abs(map[nums[i]] - i) <= k:
                return True
            map[nums[i]] = i
        return False